src/app/src/advancedsetupdialog.h

```diff
@@ -0,0 +1,39 @@
+#ifndef  ADVANCEDSETUPDIALOG_H
+#define  ADVANCEDSETUPDIALOG_H
+
+
+#include <QDialog>
+#include <QListWidget>
+#include <QListWidgetItem>
+#include <toml.hpp>
+
+#include <NIDAQmx_wrapper.h>
+
+namespace Ui {
+class AdvancedSetupDialog;
+}
+/*
+ * Nautilus advanced setup window class.
+ */
+class AdvancedSetupDialog : public QDialog{
+    Q_OBJECT
+
+    public:
+        explicit AdvancedSetupDialog(toml::value config,std::string* m_niDev, QWidget* parent = 0);
+        ~AdvancedSetupDialog();
+        void Initialize(std::vector<std::string>);
+        std::string Get_Device_Selected();
+
+    signals:
+        void sig_ni_dev_change();
```

Comment:
this should take a parameter of a `std::string` and when emitting this signal pass in the `m_niDev` member variable